### PR TITLE
UPCOMING:  Release 1.10.0

### PR DESCRIPTION
**Pull request by skoudoro**
Preparation for 1.10.0 release, targeting **~~October 11, 2024~~ December 12th, 2024**

# Desired PR / issues (need review / feedback / update)

## PR

- [x]  #3358 and  #3170  (Submitter: @samcoveney)
- [x] #3303 (Submitter: @iprajwalreddy)
- [x] #3140 (submitter: @pjsjongsung )
- [ ] #3089 (submitter: @gabknight )
- [x] #1957 (submitter: @skoudoro)
- [x] #3169 (submitter: @skoudoro)
- [x] #2826 (submitter: @CHrlS98 

## Issues/Features without PR yet

- [x] #3361
- [x] #3285
- [ ] #3190
- [x] #3115
- [x] #3030
- [x] #2757

Most of them can be done on time, I think :smile:

Please comment to flag any other issues or PR that should be addressed for this release

# Release checklist
- [x] add release_notes 1.10.0
- [x] Update changelog and api_changes
- [x] Update .mailmap
- [x] Set release number in info.py
- [x] Check the copyright years in doc/conf.py and LICENSE
- [x] Check documentation
- [x] Check Tutorial
- [x] Check unit test and USE-PRE
- [x] Update index.rst
- [x] Update Highligth.rst, old_news.rst, stateofheart.rst
- [x] Update developers.rst list
- [ ] Update website
- [x] Update deprecation cycle remove long deprecation

### CODE REVIEW

**Comment by skoudoro** [comment]
ok, https://github.com/dipy/dipy/pull/3089 and https://github.com/dipy/dipy/issues/3190 will move to the next release.

I am starting to tackle the release checklist

**Comment by skoudoro** [comment]
Can you look at this PR @arokem, @Garyfallidis and @jhlegarreta ?

Thank you

**Comment by jhlegarreta** [comment]
Looks OK to me; I would release without waiting for a PR to come for #3190. Thanks for all the work that making the release involves @skoudoro.

File: Changelog [hunk 1]
```diff
@@ -24,7 +24,24 @@ Dipy
...
+
+- NF: Patch2Self3 - Large improvements of self-supervised denoising method added.
+- NF: Fiber density and spread from ODF using Bingham distributions method added.
+- NF: Iteratively reweighted least squares for robust fitting of diffusion models added.
+- NF: NDC - Neighboring DWI Correlation quality metric added.
+- NF: DAM - tissue classification method added.
+- NF: New Parallel Backends (Ray, joblib, Dask) for fitting reconstruction methods added.
+- RF: Deprecation of Tensorflow support. PyTorch support is now the default.
+- Transition to Keyword-only arguments (PEP 3102).
+- Zero-warnings policy (CIs, Compilation, doc generation) adopted.
```
Comment (jhlegarreta) [inline Changelog:37]: - Zero-warnings policy (CIs, Compilation, doc generation) adopted.
```suggestion
- Zero-warnings policy (CIs, Compilation, doc generation) adopted.
- Adoption of ruff for automatic style enforcement
- Transition to using f-strings.
```

**Pull request merged**